tools/validation-generator/src/main/java/io/spine/tools/validate/field/NumberFieldValidatorFactory.java

```diff
@@ -0,0 +1,268 @@
+        if (boundaries.hasMin()) {
+            Boundary min = boundaries.min();
+            @SuppressWarnings("DuplicateStringLiteralInspection") // In tests.
+            Constraint rule = boundaryRule(min, "<", "<=", "greater");
+            rules.add(rule);
+        }
+        if (boundaries.hasMax()) {
+            Boundary max = boundaries.max();
+            Constraint rule = boundaryRule(max, ">", ">=", "less");
+            rules.add(rule);
+        }
+        return rules.build();
+    }
+
+    private void checkNotRequired() {
+        if (field().hasOption(required)) {
+            _warn().log(
+                    "Option (required) is not applicable to number fields. Found at %s.",
+                    field()
+            );
+        }
+    }
+
+    @Override
+    public Expression<Boolean> isNotSet() {
+        return Expression.of(valueOf(false));
+    }
+
+    @Override
+    public boolean supportsRequired() {
+        return false;
+    }
+
+    private Constraint boundaryRule(Boundary boundary,
+                                    String exclusiveOperator,
+                                    String inclusiveOperator,
+                                    String englishDescription) {
+        boolean inclusive = boundary.inclusive();
+        String operator = inclusive ? exclusiveOperator : inclusiveOperator;
+        Constraint rule = new Constraint(
+                field -> formatted("%s %s %s", field, operator, boundary.value()),
+                field -> violationTemplate()
+                        .setMessage(format("Field must be %s than%s %s.",
+                                           englishDescription,
+                                           inclusive ? " or equal to" : "",
+                                           boundary.value()))
+                        .setFieldValue(field)
+                        .build()
+        );
+        return rule;
+    }
+
+    private NumberBoundaries parseBoundaries() {
+        checkNotTooMuchOptions();
+        NumberBoundaries boundaries = minMaxBoundaries();
+        return boundaries.isBound()
+               ? boundaries
+               : rangeBoundaries();
+    }
+
+    private void checkNotTooMuchOptions() {
+        FieldDeclaration field = field();
+        checkState(!(field.hasOption(range) && (field.hasOption(min) || field.hasOption(max))),
+                   "Detected usage of (range) alongside with (min) or (max) at `%s`.", field);
+    }
+
+    private NumberBoundaries minMaxBoundaries() {
+        Boundary low = null;
+        MinOption minOption = field().findOption(min);
+        if (isNotDefault(minOption)) {
+            low = new Boundary(numberKind.parse(minOption.getValue()), !minOption.getExclusive());
+
+        }
+        Boundary high = null;
+        MaxOption maxOption = field().findOption(max);
+        if (isNotDefault(maxOption)) {
+            high = new Boundary(numberKind.parse(maxOption.getValue()), !maxOption.getExclusive());
```

Comment:
The construction of `Boundary` for `low` and `high` shares the same principle. Can it be extracted into a method?